### ISSUE TITLE
not loaded other shards for training

### ISSUE THREAD

**Issue by fermat97** [OpenNMT/OpenNMT-py#1262]
I have 5 shards, each of which almost 5 million sentences. And I set train steps 1000, and I use two GPUs. But the training is over after almost half an hour and it only shows loading the first shard. And I dont receive any error for out-of-memory issues. I use bachsize of 8. I assumed each iteration should be run on whole 5 shards, is that true? I use python Python 3.7.2 and Pytorch version is 1.0.0

    CUDA_VISIBLE_DEVICES=2,3 python -u train.py -data data/data.bpe 
                        -save_model path-to-model 
                        -src_word_vec_size 1024 
                        -tgt_word_vec_size 1024 
                        -encoder_type brnn 
                        -enc_layers 2 
                        -dec_layers 2 
                        -enc_rnn_size 2048
                        -dec_rnn_size 2048
                        -bridge 
                        -rnn_type LSTM 
                        -global_attention mlp 
                        -gpu_ranks 0 1 
                        -world_size 2 
                        -batch_size 8 
                        -valid_batch_size 8 
                        -train_steps 1000
                        -optim adam 
                        -dropout 0.3 
                        -learning_rate 0.001 
                        -log_file model.log

**Comment by vince62s** [comment]
train_steps 1000 is not what you want. start with 100000. with 5 mio segments you probably want even more. Look at the examples in the doc.
cheers.

**Comment by fermat97** [comment]
I am so sorry but I don't know which example do you mean, I think I have seen most of them but may be I missed something. But with a work around I think since I have 24339550 (total number of sentences in all shards) whit batch bachsize of 8, I have to fixed the train_steps to the value `(24339550 /8)*100 = 304244375` for 100 epochs. But I am not sure if this is also valid for multi GPUs (I work with at least 2 GPUs). I assume that if I use two GPUs then for the same amount of epochs I have  to divide it by 2 (304244375/2), I fix it as 152122187. Am I right, with all these computations?
